### ISSUE TITLE
Add published date attribute to video 

### ISSUE THREAD

**Issue by Chocobozzz** [Chocobozzz/PeerTube#234]
It's more useful for users to see when a video was published (privacy set to unlisted/public) than when it was created.

 * Add a `publishedAt` video attribute in database
 * Create a migration script that set `publishedAt` = `createdAt` for old videos (need to update the `LAST_MIGRATION_VERSION` in `constants.ts`
 * Set `publishedAt` = NOW when the video becomes public/unlisted (on upload/update)
 * Add this attribute in `VideoTorrentObject` for ActivityPub
 * Add this new field when sending the video to followers
 * Parse this new field when receiving a new video
 * Display `publishedAt` attribute instead of `createdAt` on the client interface (except in the My Account -> My videos panel)
 * (Optional) Update tests (I can do it)

**Comment by jlebras** [comment]
Hello, I would like to contribute to this issue if no one has already started working on it.
(Actually, I'm already working on it, try to stop me :D)

The `VideoTorrentObject` has already a `published` attribute which seems to hold the ISOString of the `createdAt` field. Should I reuse this attribute to avoid confusion or are they 2 distinct things ?

**Comment by Chocobozzz** [comment]
Yep, reuse this attribute to use the `publishedAt` video instead of `createdAt`. With the new attribute `publishedAt` we will not use `createdAt` anymore

**Comment by rigelk** [comment]
Implemented (partially?) in #439 - thanks @jlebras !
